lib/orderlist.rb

```diff
@@ -0,0 +1,87 @@
+require 'dotenv'
+require 'twilio-ruby'
+
+class OrderList
+  def initialize(restaurant, input = Kernel, order_class = Order, sms_class = Twilio::REST::Client)
+    @menu = restaurant.menu
+    @order_class = order_class
+    @input = input
+    @sms_class = sms_class
+    @confirmed_orders = []
+  end
+
+  def new_order
+    pending_order = @order_class.new
+    
+    loop do
+      show_choices
+
+      response = @input.gets.chomp
+      break if response == "quit"
+
+      if response == "1"
+        show_menu
+      elsif response == "2"
+        add_to_order(pending_order)
+      elsif response == "confirm"
+        finalise_order(pending_order)
+        break
+      end
+
```

Comment:
Could create a method as a condition checker for the response input variable, as it would make the new_order method cleaner and more SRP-oriented.